docs/reference/contributing/guidelines/workflow.md

```diff
@@ -61,90 +61,143 @@ If commits do not follow the above guidelines, we may request that you modify th
+    ##### Impact of changes
 
-It must contain:
+    ##### Migration actions required
 
-- A brief description of changes introduced, including justification.
-- An analysis of effects: components affected and potential consequences for users.
-- Migration guidance: actions for updating the current code. Please include code snippets to illustrate before and after the addition or change.
 
-<span class="notes">**Note:** We may use this content in our official release notes.</span>
+### Description field
 
-For more details, please see the [pull request template addition for functional changes](#pull-request-template-functional-changes).
+There are two parts to the description, both of which are required.
+The first is a summary of the pull request. This should clearly state the reason for the PR and what 
+the changes entail. The second part requires the author to state what, if any, documentation changes 
+also need to be made to accompany the changes.
 
-We initially implement new features on separate branches in the Mbed OS repository. Mbed OS maintainers create the new branches by following the naming convention: "feature-" prefix.
+### Pull request type
 
-Each feature has a tech lead. This person is responsible for:
+There are three pull request types and these correspond to the 3 main categories specified in semantic versionning, Patch, Feature (Minor) and Major.
 
-- Rebasing often to track master development.
-- Reviewing any addition to the feature branch (approval required by the feature tech lead or another assigned person).
+#### Patch update
 
-Release: feature
+This can contain any of the following changes:
 
-### Documentation update
+##### Bug Fix
 
-Documentation updates include changes to markdown files or Doxygen documentation (comment-only changes).
+A bug fix is a change that fixes a specific defect in the codebase with backward compatibility. These are the highest priority because of the positive effect the change will have on users developing against the same code. A bug fix should be limited to restoring the documented or proven otherwise, originally intended behavior. Every bug fix should contain a test to verify results before and after the pull request. Bug fixes are candidates for patch releases. Large, nontrivial bug fixes approaching the size of refactors run the risk of being considered refactors themselves.
+
+##### Refactor
 
-Release: patch
+A refactor is a contribution that modifies the codebase without fixing a bug or changing the existing behavior. Examples of this are moving functions or variables between translation units, renaming source files or folders, scope modification for nonpublic code, documentation structure changes and test organization changes. There is always the risk that someone depended on the location or name before a refactor; therefore, these are lower in priority than bug fixes and might require detailed justification for the change. Refactors are candidates for feature releases.
```

Comment:
> Refactors are candidates for feature releases.

this one should be updated to reflect PR type above - patch release

Reply:
Ah just need to remove that last sentence completely as it is already under the Patch section